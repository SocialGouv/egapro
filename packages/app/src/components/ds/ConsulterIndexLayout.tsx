import { Box, Container, Flex } from "@chakra-ui/layout";
import type { PropsWithChildren } from "react";
import React from "react";

import { Footer } from "./Footer";
import { Header } from "./Header";

// eslint-disable-next-line @typescript-eslint/ban-types -- no props
export const ConsulterIndexLayout = ({ children }: PropsWithChildren<{}>) => {
  return (
    <Flex direction="column" minHeight="100vh">
      <Header />
      <Box as="main" role="main" id="main" flexGrow={1} pt={10}>
        <Container maxW="container.lg">{children}</Container>
      </Box>
      <Footer />
    </Flex>
  );
};
