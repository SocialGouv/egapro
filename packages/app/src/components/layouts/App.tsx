import { EntrepriseFooter } from "@components/Footers";
import { ButtonAsLink, FormButton, Logo, SkipLinks, SkipLinksItem } from "@design-system";
import type { TokenInfoType } from "@services/apiClient";
import { useFormManager, useUser } from "@services/apiClient";
import { clsx } from "clsx";
import Head from "next/head";
import NextLink from "next/link";
import { useRouter } from "next/router";
import type { PropsWithChildren } from "react";
import { useCallback, useEffect, useState } from "react";

import styles from "./App.module.css";

interface ActionButtonGroupsProps {
  dest: string;
  disconnectUser: VoidFunction;
  isAuthenticated: boolean;
  user: TokenInfoType | undefined;
}

const ActionButtonGroups = ({ dest, disconnectUser, isAuthenticated, user }: ActionButtonGroupsProps) => (
  <ul className="fr-btns-group">
    {isAuthenticated ? (
      <>
        <li>
          <NextLink href="/index-egapro/tableauDeBord/mon-profil" passHref legacyBehavior>
            <ButtonAsLink iconLeft="fr-icon-account-fill">
              {user?.email}
              {user?.staff ? " (staff)" : ""}
            </ButtonAsLink>
          </NextLink>
        </li>
        <li>
          <FormButton type="button" variant="secondary" iconLeft="fr-icon-lock-fill" onClick={disconnectUser}>
            Se déconnecter
          </FormButton>
        </li>
      </>
    ) : (
      <li>
        <NextLink href={`/representation-equilibree/email?redirectTo=${dest}`} passHref legacyBehavior>
          <ButtonAsLink type="button" variant="secondary" iconLeft="fr-icon-lock-fill">
            Se connecter
          </ButtonAsLink>
        </NextLink>
      </li>
    )}
  </ul>
);

export const App = ({
  children,
  footer = <EntrepriseFooter />,
}: PropsWithChildren & { footer?: React.ReactElement }) => {
  const router = useRouter();
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const mobileMenuId = "mobile-menu";
  const buttonMobileMenuId = "button-mobile-menu";

  const { isAuthenticated, logout, user } = useUser();
  const { resetFormData } = useFormManager();

  const disconnectUser = () => {
    logout();
    resetFormData();
  };

  const setTheme = useCallback(() => {
    const askForDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
    const isDark = document.documentElement.getAttribute("data-fr-theme") === "dark";
    if (askForDark) {
      if (isDark) return;
      document.documentElement.setAttribute("data-fr-theme", "dark");
    } else {
      if (!isDark) return;
      document.documentElement.setAttribute("data-fr-theme", "light");
    }
  }, []);

  useEffect(() => {
    setTheme();
    const handleChange = () => setTheme();
    window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", handleChange);
    return function cleanup() {
      window.removeEventListener("change", handleChange);
    };
  }, [setTheme]);

  return (
    <>
      <Head>
        <meta
          name="description"
          content="Egapro permet aux entreprises de mesurer, en toute transparence, les écarts de rémunération entre les sexes et de mettre en évidence leurs points de progression."
        />
      </Head>
      <div className={styles.app}>
        <SkipLinks>
          <SkipLinksItem href="#content">Contenu</SkipLinksItem>
          <SkipLinksItem href="#header">Menu</SkipLinksItem>
          <SkipLinksItem href="#footer">Pied de page</SkipLinksItem>
        </SkipLinks>
        <header role="banner" className="fr-header" id="header">
          <div className="fr-header__body">
            <div className="fr-container">
              <div className="fr-header__body-row">
                <div className="fr-header__brand fr-enlarge-link">
                  <div className="fr-header__brand-top">
                    <div className="fr-header__logo">
                      <Logo />
                    </div>
                    <div className="fr-header__navbar">
                      <button
                        className="fr-btn--menu fr-btn"
                        data-fr-opened={isMenuOpen}
                        aria-controls={mobileMenuId}
                        aria-haspopup="menu"
                        id={buttonMobileMenuId}
                        title="Menu"
                        onClick={() => setIsMenuOpen(true)}
                      >
                        Menu
                      </button>
                    </div>
                  </div>
                  <div className="fr-header__service">
                    <NextLink href="/" title="Accueil - Egapro - Ministère du Travail, de l’Emploi et de l’Insertion">
                      <p className="fr-header__service-title">Egapro</p>
                    </NextLink>
                    <p className="fr-header__service-tagline">
                      Index de l’égalité professionnelle et représentation équilibrée femmes – hommes
                    </p>
                  </div>
                </div>
                <div className="fr-header__tools">
                  <div className="fr-header__tools-links">
                    <ActionButtonGroups
                      dest={router.pathname}
                      isAuthenticated={isAuthenticated}
                      disconnectUser={disconnectUser}
                      user={user}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className={clsx("fr-header__menu fr-modal", isMenuOpen && "fr-modal--opened")} id={mobileMenuId}>
            <div className="fr-container">
              <button
                className="fr-btn--close fr-btn"
                aria-controls={mobileMenuId}
                title="Fermer"
                onClick={() => setIsMenuOpen(false)}
              >
                Fermer
              </button>
              <div className="fr-header__menu-links">
                <ActionButtonGroups
                  dest={router.pathname}
                  isAuthenticated={isAuthenticated}
                  disconnectUser={disconnectUser}
                  user={user}
                />
              </div>
            </div>
          </div>
        </header>
        <main role="main" id="content" className={styles.content}>
          {children}
        </main>
        {footer}
      </div>
    </>
  );
};
