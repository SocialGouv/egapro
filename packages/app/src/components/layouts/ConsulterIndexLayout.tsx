import { Box, Container, Flex } from "@chakra-ui/layout";
import { ChakraProvider } from "@chakra-ui/react";
import Head from "next/head";
import type { PropsWithChildren } from "react";
import React from "react";
import { theme } from "../../theme";

import { Footer } from "../ds/Footer";
import { Header } from "@components/ds/Header";
/**
 * Layout for consulter pages.
 */
export const ConsulterIndexLayout = ({ children, title }: PropsWithChildren & { title?: string | undefined }) => {
  return (
    <ChakraProvider theme={theme}>
      <Head>
        <title>{title && title + " - "} Egapro</title>
      </Head>
      <Flex direction="column" minHeight="100vh">
        <Header />
        <Box as="main" role="main" id="main" flexGrow={1} pt={10}>
          <Container maxW="container.lg">{children}</Container>
        </Box>
        <Footer />
      </Flex>
    </ChakraProvider>
  );
};
