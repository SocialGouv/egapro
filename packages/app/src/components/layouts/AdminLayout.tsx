import { ClientOnly } from "@components/ClientOnly";
import { DsfrScript } from "@components/DsfrScript";
import { StaffOnly } from "@components/StaffOnly";
import Head from "next/head";
import type { PropsWithChildren } from "react";

import { App } from "./App";

const DEFAULT_TITLE = "Egapro Backoffice";

/**
 * Layout for admin pages like bo.
 */
export const AdminLayout = ({ children, title }: PropsWithChildren & { title?: string | undefined }) => {
  return (
    <App>
      <Head>
        <title>{title ? title + " - " + DEFAULT_TITLE : DEFAULT_TITLE}</title>
      </Head>
      <DsfrScript />

      <ClientOnly>
        <StaffOnly>{children}</StaffOnly>
      </ClientOnly>
    </App>
  );
};
