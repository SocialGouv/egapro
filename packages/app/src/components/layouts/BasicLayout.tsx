import "@gouvfr/dsfr/dist/dsfr.main.min.css";
import "@gouvfr/dsfr/dist/utility/icons/icons-system/icons-system.min.css";
import "@gouvfr/dsfr/dist/utility/icons/icons-user/icons-user.min.css";
import "@gouvfr/dsfr/dist/utility/icons/icons-business/icons-business.min.css";

import Head from "next/head";
import type { PropsWithChildren } from "react";
import { App } from "./App";

const DEFAULT_TITLE = "Egapro";

/**
 * Layout for unauthenticated users (standalone pages, etc.).
 */
export const BasicLayout = ({ children, title }: PropsWithChildren & { title?: string | undefined }) => {
  return (
    <App>
      <Head>
        <title>{title ? title + " - " + DEFAULT_TITLE : DEFAULT_TITLE}</title>
      </Head>
      {children}
    </App>
  );
};
