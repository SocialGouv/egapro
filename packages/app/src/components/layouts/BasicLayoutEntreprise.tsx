import { DsfrScript } from "@components/DsfrScript";
import { EntrepriseFooter } from "@components/Footers";
import Head from "next/head";
import type { PropsWithChildren } from "react";

import { App } from "./App";

const DEFAULT_TITLE = "Egapro";

/**
 * Layout for unauthenticated users (standalone pages, etc.).
 */
export const BasicLayoutEntreprise = ({ children, title }: PropsWithChildren & { title?: string | undefined }) => {
  return (
    <App footer={<EntrepriseFooter />}>
      <Head>
        <title>{title ? title + " - " + DEFAULT_TITLE : DEFAULT_TITLE}</title>
      </Head>
      <DsfrScript />
      {children}
    </App>
  );
};
