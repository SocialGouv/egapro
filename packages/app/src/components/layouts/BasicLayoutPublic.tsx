import { PublicFooter } from "@components/Footers";
import Head from "next/head";
import { type PropsWithChildren } from "react";

import { App } from "./App";

const DEFAULT_TITLE = "Egapro";

/**
 * Layout for unauthenticated users (standalone pages, etc.).
 */
export const BasicLayoutPublic = ({ children, title }: PropsWithChildren & { title?: string | undefined }) => {
  return (
    <App footer={<PublicFooter />}>
      <Head>
        <title>{title ? title + " - " + DEFAULT_TITLE : DEFAULT_TITLE}</title>
      </Head>
      {children}
    </App>
  );
};
