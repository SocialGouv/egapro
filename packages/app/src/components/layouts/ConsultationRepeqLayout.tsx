import { PublicFooter } from "@components/Footers";
import { Container, Grid, GridCol, SideMenuLink } from "@design-system";
import Head from "next/head";
import { useRouter } from "next/router";
import { type PropsWithChildren } from "react";

import { type AppProps } from "./App";
import { App } from "./App";

export type AsideLinkProps = PropsWithChildren<{
  path: string;
}>;

export const AsideLink = ({ path, children }: AsideLinkProps) => {
  const router = useRouter();
  const currentRoute = router.pathname;
  return (
    <SideMenuLink
      href={`/representation-equilibree/${path}/`}
      isCurrent={currentRoute === `/representation-equilibree/${path}`}
    >
      {children}
    </SideMenuLink>
  );
};

const DEFAULT_TITLE = "Recherche - Représentation équilibrée Egapro";

export interface ConsultationRepeqLayoutProps extends AppProps {
  title?: string;
}
export const ConsultationRepeqLayout = ({
  children,
  title,
  ...rest
}: PropsWithChildren<ConsultationRepeqLayoutProps>) => {
  return (
    <App footer={<PublicFooter />} {...rest}>
      <Head>
        <title>{title ? title + " - " + DEFAULT_TITLE : DEFAULT_TITLE}</title>
      </Head>

      <Container py="6w">
        <Grid justifyCenter>
          <GridCol md={10} lg={8}>
            {children}
          </GridCol>
        </Grid>
      </Container>
    </App>
  );
};
