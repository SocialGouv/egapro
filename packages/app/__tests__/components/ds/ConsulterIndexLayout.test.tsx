import { render } from "@testing-library/react";
import { ConsulterIndexLayout } from "@components/layouts/ConsulterIndexLayout";

test("should match snapshot", () => {
  const view = render(<ConsulterIndexLayout>Hello</ConsulterIndexLayout>);
  expect(view).toMatchSnapshot();
});
