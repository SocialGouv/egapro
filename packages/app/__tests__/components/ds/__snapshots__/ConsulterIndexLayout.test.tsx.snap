// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div>
      <div
        class="css-zf0iqh"
      >
        <header
          class="css-6rrd1m"
          role="banner"
        >
          <div
            class="css-9v6k7z"
          >
            <div
              class="chakra-container css-11vl7h3"
            >
              <ul
                class="chakra-stack css-9zlpoh"
                role="list"
              >
                <li
                  class="css-0"
                >
                  <a
                    class="chakra-link css-spn4bz"
                    href="#main"
                  >
                    Aller au contenu
                  </a>
                </li>
                <li
                  class="css-0"
                >
                  <a
                    class="chakra-link css-spn4bz"
                    href="#footer"
                  >
                    Aller au pied de page
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="chakra-container css-11vl7h3"
            id="menu"
          >
            <div
              class="css-1ibkps6"
            >
              <div
                class="css-whilru"
              >
                <a
                  class="chakra-link css-msbdq"
                  href="https://travail-emploi.gouv.fr/"
                  rel="noopener"
                  target="_blank"
                >
                  <img
                    alt="Aller à la page d'accueil du Ministère du travail, de l'emploi et de l'insertion"
                    class="chakra-image css-6t4orf"
                    src="/icons/marianne.svg"
                  />
                </a>
              </div>
              <div
                class="css-1fm0la2"
              >
                <a
                  class="chakra-link css-1wvu6nm"
                  href="/"
                >
                  Egapro
                </a>
                <p
                  class="chakra-text css-dw5ttn"
                >
                  Index de l’égalité professionnelle et représentation équilibrée
                </p>
              </div>
            </div>
          </div>
        </header>
        <main
          class="css-opostm"
          id="main"
          role="main"
        >
          <div
            class="chakra-container css-14vcoqp"
          >
            Hello
          </div>
        </main>
        <footer
          class="css-1aolk1p"
          id="footer"
          role="contentinfo"
        >
          <div
            class="chakra-container css-140na8g"
          >
            <div
              class="css-v3i7xg"
            >
              <div
                class="css-1ffgthf"
              >
                <a
                  class="chakra-link css-9tv0aq"
                  href="https://travail-emploi.gouv.fr/"
                  rel="noopener"
                  target="_blank"
                >
                  <img
                    alt="Aller à la page d'accueil du Ministère du travail, de l'emploi et de l'insertion"
                    class="chakra-image css-6t4orf"
                    src="/icons/marianne.svg"
                  />
                </a>
              </div>
              <ul
                class="css-11ji3cx"
                role="list"
              />
            </div>
            <div
              class="css-1tqkm5s"
            >
              <p
                class="chakra-text css-ltugf9"
              >
                Index Egapro et Représentation équilibrée sont développés et maintenus par les équipes de la fabrique numérique des ministères sociaux.
              </p>
              <div
                class="chakra-stack css-14r7i8g"
              >
                <span
                  class="chakra-text css-itvw0n"
                >
                  Contact: 
                  <a
                    class="chakra-link css-17hbvgh"
                    href="mailto:index@travail.gouv.fr"
                    rel="noopener"
                    target="_blank"
                  >
                    index@travail.gouv.fr
                  </a>
                </span>
                <span
                  class="chakra-text css-itvw0n"
                >
                  <a
                    class="chakra-link css-17hbvgh"
                    href="https://github.com/SocialGouv/egapro/commit/<githubSha>"
                    rel="noopener"
                    target="_blank"
                  >
                    Contribuer sur GitHub
                  </a>
                </span>
              </div>
            </div>
          </div>
        </footer>
      </div>
      <span
        hidden=""
        id="__chakra_env"
      />
    </div>
    <div
      class="chakra-portal"
    >
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-top-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-left"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        aria-live="polite"
        id="chakra-toast-manager-bottom-right"
        role="region"
        style="position: fixed; z-index: var(--toast-z-index, 5500); pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="css-zf0iqh"
    >
      <header
        class="css-6rrd1m"
        role="banner"
      >
        <div
          class="css-9v6k7z"
        >
          <div
            class="chakra-container css-11vl7h3"
          >
            <ul
              class="chakra-stack css-9zlpoh"
              role="list"
            >
              <li
                class="css-0"
              >
                <a
                  class="chakra-link css-spn4bz"
                  href="#main"
                >
                  Aller au contenu
                </a>
              </li>
              <li
                class="css-0"
              >
                <a
                  class="chakra-link css-spn4bz"
                  href="#footer"
                >
                  Aller au pied de page
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="chakra-container css-11vl7h3"
          id="menu"
        >
          <div
            class="css-1ibkps6"
          >
            <div
              class="css-whilru"
            >
              <a
                class="chakra-link css-msbdq"
                href="https://travail-emploi.gouv.fr/"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Aller à la page d'accueil du Ministère du travail, de l'emploi et de l'insertion"
                  class="chakra-image css-6t4orf"
                  src="/icons/marianne.svg"
                />
              </a>
            </div>
            <div
              class="css-1fm0la2"
            >
              <a
                class="chakra-link css-1wvu6nm"
                href="/"
              >
                Egapro
              </a>
              <p
                class="chakra-text css-dw5ttn"
              >
                Index de l’égalité professionnelle et représentation équilibrée
              </p>
            </div>
          </div>
        </div>
      </header>
      <main
        class="css-opostm"
        id="main"
        role="main"
      >
        <div
          class="chakra-container css-14vcoqp"
        >
          Hello
        </div>
      </main>
      <footer
        class="css-1aolk1p"
        id="footer"
        role="contentinfo"
      >
        <div
          class="chakra-container css-140na8g"
        >
          <div
            class="css-v3i7xg"
          >
            <div
              class="css-1ffgthf"
            >
              <a
                class="chakra-link css-9tv0aq"
                href="https://travail-emploi.gouv.fr/"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="Aller à la page d'accueil du Ministère du travail, de l'emploi et de l'insertion"
                  class="chakra-image css-6t4orf"
                  src="/icons/marianne.svg"
                />
              </a>
            </div>
            <ul
              class="css-11ji3cx"
              role="list"
            />
          </div>
          <div
            class="css-1tqkm5s"
          >
            <p
              class="chakra-text css-ltugf9"
            >
              Index Egapro et Représentation équilibrée sont développés et maintenus par les équipes de la fabrique numérique des ministères sociaux.
            </p>
            <div
              class="chakra-stack css-14r7i8g"
            >
              <span
                class="chakra-text css-itvw0n"
              >
                Contact: 
                <a
                  class="chakra-link css-17hbvgh"
                  href="mailto:index@travail.gouv.fr"
                  rel="noopener"
                  target="_blank"
                >
                  index@travail.gouv.fr
                </a>
              </span>
              <span
                class="chakra-text css-itvw0n"
              >
                <a
                  class="chakra-link css-17hbvgh"
                  href="https://github.com/SocialGouv/egapro/commit/<githubSha>"
                  rel="noopener"
                  target="_blank"
                >
                  Contribuer sur GitHub
                </a>
              </span>
            </div>
          </div>
        </div>
      </footer>
    </div>
    <span
      hidden=""
      id="__chakra_env"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
